ad_rando/seed_generator.py

```diff
@@ -0,0 +1,157 @@
+        'randomToolkit': ('RM3T #3 Random Toolkit Tournament', ManualRandomizerParams('dE:-2,fh:1,iInS:0,txX'))
+    }
+
+    @classmethod
+    def select(cls, searched_preset_name):
+        def matches_preset(preset):
+            preset_name, _ = preset
+            return preset_name.startswith(searched_preset_name)
+
+        def matches_preset_exactly(preset):
+            preset_name, _ = preset
+            return preset_name == searched_preset_name
+
+        matching_presets = list(filter(matches_preset, cls.all_presets().items()))
+        if len(matching_presets) > 1:
+            matching_presets = list(filter(matches_preset_exactly, matching_presets))
+        if len(matching_presets) != 1:
+            raise CurryError("That doesn't look like a seed or preset. Curry.")
+        matching_preset = matching_presets[0]
+        _, descriptor = matching_preset
+        return descriptor
+
+    @classmethod
+    def all_presets(cls):
+        presets = cls._adrando_presets()
+        presets.update(cls.MANUAL_PRESETS)
+        return presets
+
+    @classmethod
+    def _adrando_presets(cls):
+        return dict(
+            (preset_name, (description, FromPresetRandomizerParams(preset_name)))
+            for preset_name, description
+            in cls.ADRANDO_PRESETS.items())
+
+
+class RandoCommandHandler:
+    RANDO_LINKS = [SeedGenerator.RANDO_BASE]
+
+    def __init__(self, ctx, args):
+        self._ctx = ctx
+        self._args = args
+
+    async def handle(self):
+        if len(self._args) == 0:
+            await self._print_current_rando_seed_links()
+            return
+        preset_name = self._args[0]
+        if preset_name.startswith('preset'):
+            await self._print_presets()
+        else:
+            await self._generate_and_print_seeds(preset_name)
+
+    async def _print_current_rando_seed_links(self):
+        await self._ctx.send(curry_message("Current rando seed links:"))
+        await self._ctx.send(
+            '\n'.join(
+                curry_message(f"Seed {i + 1}: <{link}>")
+                for i, link
+                in enumerate(self.RANDO_LINKS)))
+
+    async def _print_presets(self):
+        await self._ctx.send(
+            '\n'.join(
+                curry_message(f"Preset: {preset}, Description: {description}")
+                for preset, (description, _)
+                in AdRandomizerParamsDescriptorSelector.all_presets().items()))
+
+    async def _generate_and_print_seeds(self, preset_name):
+        try:
+            params_descriptor = AdRandomizerParamsDescriptorSelector.select(preset_name)
+            description, params = params_descriptor
+            seeds_number = self._parse_seeds_number()
+            await ctx.send(curry_message(f"Generating {seeds_number} {description} seeds..."))
+            RandoCommandHandler.RANDO_LINKS = SeedGenerator(params, seeds_number).generate()
+            await ctx.send(curry_message("Rando seed links updated"))
```

Comment:
As discussed, try to cut down to only one 'ctx.send' being performed in curr_bot.py, if you can

Reply:
Fixed in the latest commit.